### PR TITLE
Feature/backend convert to typescript

### PR DESCRIPTION
**Pull request by snaanortey**
Converted the following to typescript:
1. `get ingredients by recipes id` API
2. `image search` API that returns an array of ingredients on the image
3.  `ingredients to recipes` API that takes the ingredients array and returns recipes that match the ingredients
4. `search and suggest meals` API that takes search suggestions and returns a list of meals whose prefixes match the suggestions
5. function for populating elastic search database with the content from JSON file

### CODE REVIEW

File: server/src/routes/searchAndSuggestMeals.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+      _source: false,
+      query: {
+        match_phrase_prefix: {
+          title: req.query.query,
+        },
+      },
+      fields: ["title"],
+    };
+
+    const result = await elasticClient.search<ElasticSearchIndexRecipe>(query);
```
Comment (ehimah) [inline server/src/routes/searchAndSuggestMeals.ts:45]: this logic should be abstracted into the `services/elasticClient` module.

You should create a function that takes an elastic search query as parameter, and returns the search result object as its result.

File: server/src/services/search.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+interface PartialMatchQueries {
+  match: PartialMatchQueriesInnerValue;
+}
+
+interface PartialMatchQueriesInnerValue {
+  ingredients: string;
+}
+
+
+
```
Comment (ehimah) [inline server/src/services/search.ts:14]: remove extra space

File: server/src/routes/getIngredientsById.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+      index: "recipes",
+      id: req.params.id,
+    };
+
+    const result = await elasticClient.get(query);
+
+    res.json(result._source);
+  } catch (err) {
+    console.log("Error getting ingredients by mealId", err);
+    // if err is an object of the class ResponseError
```
Comment (ehimah) [inline server/src/routes/getIngredientsById.ts:19]: the parameter is called id not `mealId`
```suggestion
    console.log("Error getting ingredients by id", err);
    // if err is an object of the class ResponseError
```

File: server/src/index.ts [hunk 4]
```diff
@@ -1,3 +1,30 @@
...
+import fileUpload from "express-fileupload";
+import "express-async-errors";
+import { config } from "dotenv";
+import bodyParser from "body-parser";
 
-console.log(greeting);
\ No newline at end of file
+config();
+
+import getIngredientsById from "../src/routes/getIngredientsById";
```
Comment (ehimah) [inline server/src/index.ts:10]: remove spaces

**Review by ehimah — CHANGES_REQUESTED** [review]
This is quite good. Just some tidbits and we'd be ready to ✅

File: server/.gitignore [hunk 5]
```diff
@@ -1 +1,3 @@
-node_modules
\ No newline at end of file
+node_modules
+.env
+dist
```
[the remark this answers is not in the document]
Comment (snaanortey) [reply server/.gitignore:3]: There are no spaces as dicussed

File: server/src/routes/imageSearch.ts [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+  try {
+    if (!req.files) {
+      res.send({
+        status: false,
+        message: "No file uploaded",
+      });
+      return;
+    }
+
+    const image = req.files.image as fileUpload.UploadedFile;
```
[the remark this answers is not in the document]
Comment (snaanortey) [reply server/src/routes/imageSearch.ts:22]: Addressed comment and committed changes

File: server/src/routes/searchAndSuggestMeals.ts (hunk 1 above)
Comment (snaanortey) [reply server/src/routes/searchAndSuggestMeals.ts:45]: maintained initial code as dicussed

File: server/src/services/search.ts (hunk 2 above)
Comment (snaanortey) [reply server/src/services/search.ts:14]: removed extra spaces and committed changes

File: server/src/routes/getIngredientsById.ts (hunk 3 above)
Comment (snaanortey) [reply server/src/routes/getIngredientsById.ts:19]: updated from `mealId` to `id` and committed changes

File: server/src/index.ts (hunk 4 above)
Comment (snaanortey) [reply server/src/index.ts:10]: removed extra spaces and committed changes

**Comment by snaanortey** [comment]
Thanks for the comments! I have responded to each of your comments and created a new commit to updated changes.

**Review by ehimah — APPROVED** [review]

**Pull request merged**